source/progress.ts

```diff
@@ -1,96 +1,95 @@
-'use strict';
-const {Transform} = require('stream');
-
-module.exports = {
-	download(response, emitter, downloadBodySize) {
-		let downloaded = 0;
-
-		return new Transform({
-			transform(chunk, encoding, callback) {
-				downloaded += chunk.length;
-
-				const percent = downloadBodySize ? downloaded / downloadBodySize : 0;
-
-				// Let `flush()` be responsible for emitting the last event
-				if (percent < 1) {
-					emitter.emit('downloadProgress', {
-						percent,
-						transferred: downloaded,
-						total: downloadBodySize
-					});
-				}
+import { Transform } from 'stream';
+import { EventEmitter } from 'events';
+
+export function download(response: any, emitter: EventEmitter, downloadBodySize?: number) {
```

Comment:
Can you find a proper type for the `response` object? Should be in the NodeJS types.

Reply:
You could use a default value for `downloadBodySize?: number`.

Reply:
I believe it's `import {IncomingMessage} from 'http'`;